cranelift/filetests/src/test_run.rs

```diff
@@ -57,11 +58,16 @@ fn is_isa_compatible(
     // since we won't be able to natively execute machine code.
     let host_arch = host.triple().architecture;
     let requested_arch = requested.triple().architecture;
-    if host_arch != requested_arch {
-        return Err(format!(
-            "skipped {}: host can't run {:?} programs",
-            file_path, requested_arch
-        ));
+
+    match (host_arch, requested_arch) {
+        (host, requested) if host == requested => {}
+        (Architecture::Riscv64(_), Architecture::Riscv64(_)) => {}
```

Comment:
Why do we ignore the argument to the `Riscv64` enum arm here? If we really need to special-case RISC-V in the toplevel test runner, can we factor the logic out somewhere else (to a "compatible RISC-V variant" helper function or something like that)?

Reply:
the current implementation is a little sloppy.
~~~
fn is_riscv64_compatible(
+    host: target_lexicon::Riscv64Architecture,
+    req: target_lexicon::Riscv64Architecture,
+) -> bool {
+    match host {
+        // Riscv64gc is short for RV64IMAFDCZicsr_Zifencei.
+        // So can run them all.
+        target_lexicon::Riscv64Architecture::Riscv64gc => true,
+        // Riscv64imac can run when req is not Riscv64gc.
+        target_lexicon::Riscv64Architecture::Riscv64imac
+            if req != target_lexicon::Riscv64Architecture::Riscv64gc =>
+        {
+            true
+        }
+        // Riscv64 is just basic extension.
+        target_lexicon::Riscv64Architecture::Riscv64
+            if req == target_lexicon::Riscv64Architecture::Riscv64 =>
+        {
+            true
+        }
+        _ => false,
+    }
+}
+
 /// Checks if the host's ISA is compatible with the one requested by the test.
 fn is_isa_compatible(
     file_path: &str,
@@ -61,7 +85,8 @@ fn is_isa_compatible(
 
     match (host_arch, requested_arch) {
         (host, requested) if host == requested => {}
-        (Architecture::Riscv64(_), Architecture::Riscv64(_)) => {}
+        (Architecture::Riscv64(host), Architecture::Riscv64(req))
+            if is_riscv64_compatible(host, req) => {}
         _ => {
             return Err(format!(
                 "skipped {}: host can't run {:?} programs",
~~~